File: GeneratedEntityReaderClasses/Components/ItemChestComponent.cs
namespace GeneratedNoitaClasses;
public class ItemChestComponent : NoitaComponentBase,  iNoitaType<ItemChestComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt item_count_min { get; set; } = new();
	public NoitaInt item_count_max { get; set; } = new();
	public NoitaInt level { get; set; } = new();
	public NoitaBool enemy_drop { get; set; } = new();
	public NoitaString actions { get; set; } = new();
	public NoitaString action_uses_remaining { get; set; } = new();
	public NoitaString other_entities_to_spawn { get; set; } = new();
	public NoitaUInt mSeed { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		item_count_min.Read(s);
		item_count_max.Read(s);
		level.Read(s);
		enemy_drop.Read(s);
		actions.Read(s);
		action_uses_remaining.Read(s);
		other_entities_to_spawn.Read(s);
		mSeed.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		item_count_min.Write(s);
		item_count_max.Write(s);
		level.Write(s);
		enemy_drop.Write(s);
		actions.Write(s);
		action_uses_remaining.Write(s);
		other_entities_to_spawn.Write(s);
		mSeed.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ItemChestComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class ItemChestComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    item_count_min:NoitaInt = field(default_factory=NoitaInt)
    item_count_max:NoitaInt = field(default_factory=NoitaInt)
    level:NoitaInt = field(default_factory=NoitaInt)
    enemy_drop:NoitaBool = field(default_factory=NoitaBool)
    actions:NoitaString = field(default_factory=NoitaString)
    action_uses_remaining:NoitaString = field(default_factory=NoitaString)
    other_entities_to_spawn:NoitaString = field(default_factory=NoitaString)
    mSeed:NoitaUInt = field(default_factory=NoitaUInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.item_count_min.Read(s)
        self.item_count_max.Read(s)
        self.level.Read(s)
        self.enemy_drop.Read(s)
        self.actions.Read(s)
        self.action_uses_remaining.Read(s)
        self.other_entities_to_spawn.Read(s)
        self.mSeed.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.item_count_min.Write(s)
        self.item_count_max.Write(s)
        self.level.Write(s)
        self.enemy_drop.Write(s)
        self.actions.Write(s)
        self.action_uses_remaining.Write(s)
        self.other_entities_to_spawn.Write(s)
        self.mSeed.Write(s)

